public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -148,6 +159,80 @@ function legacyCreateSpanLinkFactory(
     let query: DataQuery | undefined;
     let tags = '';
 
+    // Get profiles links
+    if (config.featureToggles.traceToProfiles) {
```

Comment:
So do we have to still have the code here? I thought the trace view will handle links from dataFrame. We should aim to remove the dataLink creation code from the visualization and not add more here.

Reply:
Because it should not be here. This is tightly coupling the visualization to the data source which is bad design and it should have been changed. The data source have all this information for creating the links and it can just easily attach them to the dataFrame now (we didn't have that when first log links were added) which makes things much less coupled and is more standard how other visualizations work.